src/de/deserializer.rs

```diff
@@ -473,14 +473,19 @@ impl<'de, 'a, F: Flavor<'de>> de::Deserializer<'de> for &'a mut Deserializer<'de
     #[inline]
     fn deserialize_tuple_struct<V>(
         self,
-        _name: &'static str,
+        name: &'static str,
         len: usize,
         visitor: V,
     ) -> Result<V::Value>
     where
         V: Visitor<'de>,
     {
-        self.deserialize_tuple(len, visitor)
+        if name.as_ptr() == crate::byte_array::TOKEN.as_ptr() {
```

Comment:
I'm not certain this is *guaranteed* to be true, e.g. the linker or compiler may be free to create multiple instances of the `TOKEN` static. If you can cite rustc or llvm references that say that this is guaranteed, happy to be proven wrong.

Reply:
From https://doc.rust-lang.org/std/keyword.static.html:
> On the surface, `static` items seem very similar to [`const`](https://doc.rust-lang.org/std/keyword.const.html)s: both contain a value, both require type annotations and both can only be initialized with constant functions and values. However, `statics` are notably different in that they **represent a location in memory**. That means that you can have references to `static` items and potentially even modify them, making them essentially global variables.

It'd be catastrophic if rustc generated multiple global variables for the same one. The standard library uses it extensively:
```
$ rg 'static.*Atomic' library | wc -l
96
```

Reply:
I'm probably inventing a fear that doesn't exist, but *I could imagine* an optimization that allows for immutible statics to be replaced with consts.

`static mut`, and types that contain `UnsafeCell` like `Atomic*` are different, as mutability means the compiler cannot assume they contain the same value for the entire program lifetime.

That being said, I have no proof that this optimization exists, so I definitely wouldn't reject the PR on this point. Just sharing an anxious thought I had :)

Reply:
> All references to the static refer to the same memory location.

https://doc.rust-lang.org/reference/items/static-items.html

Nevermind then!